server/cmwell-dc/src/main/scala/cmwell/dc/stream/InfotonRetriever.scala

```diff
@@ -96,33 +111,25 @@ object InfotonRetriever extends LazyLogging {
               .via(
                 Framing.delimiter(endln, maximumFrameLength = maxStatementLength)
               )
-              .fold(
-                RetrieveTotals(
-                  state._1
-                    .map(im => (im.meta.path, (new ByteStringBuilder, None)))
-                    .toMap,
-                  new ByteStringBuilder
-                )
-              ) { (totals, nquad) =>
+              .fold(RetrieveTotals(HashMap(state._1 map{im => im.meta.path -> (new ByteStringBuilder, ExtraData("", None, None))}: _*), new ByteStringBuilder))
+              { (totals, nquad) =>
                 totals.unParsed ++= nquad
-                val parsed: Option[ParsedNquad] = parseNquad(remoteUri, nquad)
-                parsed.fold {
-                  RetrieveTotals(totals.parsed, totals.unParsed)
-                } { p =>
-                  totals.parsed.get(p.path) match {
+                val parsed: ParsedNquad = parseNquad(remoteUri, nquad)
+                parsed match {
+                  case ParsedNquad("", _, _) => RetrieveTotals(totals.parsed, totals.unParsed)
+                  case ParsedNquad(path, nquad, extraData) =>
+                    totals.parsed.get(path) match {
                     case None =>
-                      throw WrongPathGotException(
-                        s"Got path ${p.path} from _out that was not in the uuids request bulk: ${
-                          state._1
-                            .map(i => i.meta.uuid.utf8String + ":" + i.meta.path)
-                            .mkString(",")
-                        }"
-                      )
-                    case Some((builder, _)) => {
-                      builder ++= (p.nquad ++ endln)
-                      val parsedTotals = p.indexTime.fold(totals.parsed)(
-                        _ => totals.parsed + (p.path -> (builder, p.indexTime))
-                      )
+                      throw WrongPathGotException(s"Got path ${path} from _out that was not in the uuids request bulk: ${
+                        state._1.map(i => i.meta.uuid.utf8String + ":" + i.meta.path).mkString(",")}")
+                    case Some((builder, curExtraData)) => {
+                      builder ++= (nquad ++ endln)
+                      val parsedTotals = {
+                        val merged = curExtraData.shouldMerge(extraData)
+                        merged.fold(totals.parsed)
```

Comment:
You don't use the returned value. This statement has side effects only - please use `foreach` instead.